Fix: add disabled state to button

amymc opened the pull request.

dariocravero commented:
Well spotted @amymc, thanks for the PR! I'll comment on the code

dariocravero commented on __tests__/views/Action.view:
```diff
@@ -1,5 +1,6 @@
 Vertical
 onClick props.clickMe
+disabled props.isDisabled ? props.isDisabled : ''
```
I notice that you're passing `disabled` to the `Vertical` which works because even though we've ignored it until your PR, `disabled` is the prop that both web and native use underneath.

However, I've been finding the convention to start boolean props with `is` to be quite handy and you're using it there too with `props.isDisabled`.

How would you feel about us calling the prop `isDisabled` instead of `disabled` in `Vertical` and `Horizontal`?

That kind of remapping would then happen [here](https://github.com/viewsdx/morph/blob/master/morph/react-dom/get-value-for-property.js). [Here's an example](https://github.com/viewsdx/morph/blob/master/morph/react-dom/get-value-for-property.js#L10-L13) in which we map `source` in an `Image` block to `src` in the `img` tag for web.

amymc replied:
Hi @dariocravero, 

Why do we need to map from `isDisabled` to `disabled`, won't [this line](https://github.com/amymc/morph/blob/e0a03d20d7eb9ce1f67de6cbcc2d61fe0055746b/morph/react-native/block-wrap.js#L29) cover that?

dariocravero commented on __tests__/views/Action.view:
```diff
@@ -1,5 +1,6 @@
 Vertical
 onClick props.clickMe
+disabled props.isDisabled ? props.isDisabled : ''
```
As for the line of code itself on the test, it would probably read `isDisabled props.isDisabled` or `isDisabled true` (for a fixed value). We can drop the ternary here since `props.isDisabled` is casted as a boolean in any case.

dariocravero commented on morph/react-native/get-block-name.js:
```diff
@@ -57,6 +57,11 @@ const getGroupBlockName = (node, state) => {
 
+  if (hasProp(node, 'disabled')) {
+    const propNode = getProp(node, 'disabled')
+    node.disabled = propNode.value.value
```
Ideally we wouldn't store stuff inside `node`. I know that below there's `backgroundImage` but that probably shouldn't be there either.

How about as a rule of thumb we only store things in `node` if they are going to be used across a few other places or if some children need it to perform more calculations?

dariocravero commented on morph/react-native/block-wrap.js:
```diff
@@ -22,6 +22,7 @@ export const enter = (node, parent, state) => {
       `<${block}
           activeOpacity={0.7}
           onPress=${wrap(node.action)}
+          ${node.disabled ? `disabled=${wrap(node.disabled)}` : ''}
```
From [this comment below](https://github.com/viewsdx/morph/pull/16/files#r151098778), perhaps we would want to get the `isDisabled` prop here instead?

amymc commented:
Hi @dariocravero,

I've addressed most of your comments. I removed the remapping from `isDisabled` to `disabled` because it was causing problems when I tried to use it in my app. Basically in my `Login` component I'm passing the `isDisabled` prop to my `Button` component like so:
```
Button
isDisabled props.buttonDisabled
```
but in my `Button` component this prop was not available as `prop.isDisabled`, instead it was `prop.disabled`

In any case I think the remapping isn't needed because that's taken care of [here](https://github.com/amymc/morph/blob/0d919329646f48459dff45912eb73c855b72e25e/morph/react-native/block-wrap.js#L29).

What do you think?

dariocravero commented:
Amazing @amymc! Thanks! :)

dariocravero commented:
Released as v0.6.16!
